### ISSUE TITLE
rTorrent HTTP mechanism doesn't successfully pass label to client - breaks post processing

### ISSUE THREAD

**Issue by Arathen** [nosmokingbandit/Watcher3#21]
I'm not sure this is the fault of Watcher, and could just be a quirk of rtorrent. The process is different for magnets vs torrents, but the end result is the same. This only applies to snatches using rtorrent HTTP - not scgi.

For magnets:
Watcher seems to be sending the GUID.meta to the client successfully, and the submission does seem to contain the label correctly. You can briefly see the GUID download in the UI with a status of "Queued", and it does have a Watcher label on it. The problem then happens when rtorrent does its magic to convert that GUID.meta into a proper titled and active download. At that point when the file starts downloading, the label disappears.

For torrents:
Watcher sends the .torrent to the client, and it goes straight into active download. It obviously doesn't need to convert the .meta, so that step gets skipped. Unfortunately, even though the process is more straight forward, the download still has no Watcher label on it for some reason.

As a result, regardless of whether its a magnet or torrent source, once the download is complete with no label on it the post processing fails.

There isn't anything significant in the logs unfortunately. Everything looks successful as far as Watcher is concerned.

Like I said, this may be inherent quirkiness of rtorrent rather than Watcher, but it does limit the usability of rtorrent / HTTP as a download mechanism.

rtorrent / libtorrent : 0.9.6 / 0.13.6

**Comment by nosmokingbandit** [comment]
There was a problem with the bencoding module I was using. As of f57eebf125fb4cb43843daac644bbb91705b35e9 I've switched to a module that actually works.

Unfortunately, ruTorrent's httprpc only works when the stars align so I can't thoroughly test it, but take a crack at it after updating and let me know what happens.

**Comment by Arathen** [comment]
No joy, sorry :( The same result as before. Magnets add successfully as a hash meta with a correct label, before quickly re-adding as a torrent and losing the label. Torrents add successfully as a torrent, but with no label.

Theres nothing in the logs that I can give you to help unfortunately. Some additional debug messages which shed some light on what data is actually being sent/received from the downloader may help?

Thanks.

**Comment by nosmokingbandit** [comment]
I just made a bunch of little tweaks that might fix this (specifically c254cf186b1350ad120093bc81cbafe9592258e2). It works for me most of the time, but other times rTorrent doesn't load the torrent at all and the log doesn't have any useful information. I think the rTorrent dev is religiously opposed to documentation, so its tough to troubleshoot problems.

**Comment by Arathen** [comment]
Yeah, rtorrent documentation is the bane of my life right now. Trying to understand how things work is frustrating as hell.

I've updated and tested HTTP RPC, but unfortunately for me, there doesnt appear to be any improvement in behaviour. Magnets still add with a label, and then lose the label when the torrent is re-added. Torrents just add straight in with no label.

The only change I can see is that they all add in "Stopped" state now, which requires a manual start to get any action.

Sorry to not be able to help more with this :( I do appreciate the effort though.

**Comment by nosmokingbandit** [comment]
This seems to be on rTorrent/ruTorrent's end. Both the scgi and http method of sending torrents are almost identical and the scgi labels stick. What is odd is that the labels are applied, but then removed... Unfortunately rTorrent's log doesn't indicate anything regarding labels.
